wavelength-plugins/psgo.js

```diff
@@ -0,0 +1,545 @@
+				let usedPacks = {};
+				let userPacks = Db.userpacks.get(user.userid, []);
+				for (let i = 0; i < userPacks.length; i++) {
+					if (!usedPacks[userPacks[i]]) {
+						usedPacks[userPacks[i]] = 1;
+					} else {
+						usedPacks[userPacks[i]]++;
+					}
+				}
+				return this.sendReplyBox(`<strong>Your unopened packs</strong>:<br/>` + Object.keys(usedPacks).map(pack => {
+					return `<button class="button" name="send" value="/psgo packs open ${pack}">Open ${pack} Pack</button> (${usedPacks[pack]} Remaining.)`;
+				}).join('<br/>'));
+			},
+
+			list: function (target, room, user) {
+				if (!this.runBroadcast()) return;
+				this.sendReplyBox(`<div style="max-height: 300px; overflow-y: scroll;"><strong>PSGO Packs</strong><br/><br/>${packs.join(`<br/>`)}</div>`);
+			},
+			listhelp: ['/psgo packs list - displays all psgo packs'],
+		},
+
+		ladder: function (target, room, user) {
+			if (!this.runBroadcast()) return;
+			let values = {Common: 1, Uncommon: 3, Rare: 6, "Ultra Rare": 10, Legendary: 15, Mythic: 20};
+			let keys = Db.cards.keys().map(name => {
+				let points = 0;
+				let userCards = Db.cards.get(name, []);
+				if (userCards.length) {
+					for (let c = 0; c < userCards.length; c++) {
+						points += values[userCards[c].rarity] || 1;
+					}
+				}
+				return {name: name, points: points};
+			});
+			keys.sort(function (a, b) { return b.points - a.points; });
+			return this.sendReplyBox(rankLadder('PSGO Card Ladder', 'Points', keys, 'points'));
+		},
+		ladderhelp: ['/psgo ladder - show the PSGO card point ladder.'],
+
+		nuke: 'reset',
+		reset: function (target, room, user) {
+			if (!this.can('lockdown')) return;
+			if (!toId(target) || !user.psgoResetCode) {
+				let chars = `abcdefghijklmnopqurstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!@#$%^&*`.split(``);
+				let out = ``;
+				for (let i = 0; i < 10; i++) {
+					out += chars[Math.floor(Math.random() * chars.length)];
+				}
+				user.psgoResetCode = out;
+				return this.sendReplyBox(`<h1 style="color: red">WARNING</h1>You are about to remove <strong>ALL users cards AND PACKS</strong><br/>If you are <strong>100%</strong> sure you want to do this, please use:<br/><br/>/psgo reset ${user.psgoResetCode}<br/><h3 style="color: red">Once done, this is irreversible please be 100% sure!</h3>`);
+			}
+			if (user.psgoResetCode !== target.trim()) return this.parse(`/psgo reset`);
+
+			let cardKeys = Db.cards.keys();
+			for (let i = 0; i < cardKeys.length; i++) {
+				Db.cards.remove(cardKeys[i]);
+			}
+			let packKeys = Db.userpacks.keys();
+			for (let i = 0; i < packKeys.length; i++) {
+				Db.userpacks.remove(packKeys[i]);
+			}
+			Rooms.rooms.forEach(r => {
+				r.addRaw(`<div class="broadcast-red"><strong>The PSGO database was reset</strong><br/>You no longer have any cards or packs.`).update();
+			});
+		},
+		resethelp: ['/psgo reset - Wipe the PSGO database (Take ALL users cards AND packs). Requires: ~'],
```

Comment:
Nitpicky but user's

Reply:
I don't think thats correct in this case actually, since were not refering to one user.